File: server/routes/recovery.js
const { Router } = require("express");
const { v4: uuid } = require("uuid");
const db = require("../db");
const { authenticate } = require("../auth");

const router = Router();

router.get("/plans", (req, res) => {
  const plans = db.prepare("SELECT * FROM recovery_plans").all();
  res.json(plans.map((p) => ({
    id: p.id,
    name: p.name,
    description: p.description,
    duration_days: p.duration_days,
  })));
});

router.post("/enroll/:planId", authenticate, (req, res) => {
  const plan = db.prepare("SELECT id FROM recovery_plans WHERE id = ?").get(req.params.planId);
  if (!plan) return res.status(404).json({ detail: "Plan not found" });

  const id = uuid();
  db.prepare(
    "INSERT INTO patient_recoveries (id, patient_id, plan_id) VALUES (?, ?, ?)"
  ).run(id, req.user.id, plan.id);

  const pr = db.prepare("SELECT * FROM patient_recoveries WHERE id = ?").get(id);
  res.json({ patient_recovery_id: pr.id, start_date: pr.start_date });
});

router.get("/my-tasks", authenticate, (req, res) => {
  const pr = db.prepare(
    "SELECT * FROM patient_recoveries WHERE patient_id = ? AND active = 1 LIMIT 1"
  ).get(req.user.id);
  if (!pr) return res.json([]);

  const done = new Set(
    db.prepare("SELECT recovery_task_id FROM recovery_progress WHERE patient_recovery_id = ?")
      .all(pr.id)
      .map((r) => r.recovery_task_id)
  );

  const tasks = db.prepare(
    "SELECT * FROM recovery_tasks WHERE plan_id = ? ORDER BY day_number"
  ).all(pr.plan_id);

  res.json(tasks.map((t) => ({
    id: t.id,
    day_number: t.day_number,
    kind: t.kind,
    instruction: t.instruction,
    completed: done.has(t.id),
  })));
});

router.post("/complete/:taskId", authenticate, (req, res) => {
  const pr = db.prepare(
    "SELECT * FROM patient_recoveries WHERE patient_id = ? AND active = 1 LIMIT 1"
  ).get(req.user.id);
  if (!pr) return res.status(404).json({ detail: "No active recovery plan" });

  const exists = db.prepare(
    "SELECT id FROM recovery_progress WHERE patient_recovery_id = ? AND recovery_task_id = ?"
  ).get(pr.id, req.params.taskId);

  if (!exists) {
    db.prepare(
      "INSERT INTO recovery_progress (id, patient_recovery_id, recovery_task_id) VALUES (?, ?, ?)"
    ).run(uuid(), pr.id, req.params.taskId);
  }

  res.json({ status: "completed", task_id: req.params.taskId });
});

module.exports = router;

File: backend/app/routers/recovery.py
"""Post-injury / post-op recovery tracking."""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from ..database import get_db
from ..models import (
    RecoveryPlan, RecoveryTask, PatientRecovery, RecoveryProgress, User,
)
from ..schemas import RecoveryTaskOut
from ..security import get_current_user

router = APIRouter(prefix="/api/recovery", tags=["recovery"])


@router.get("/plans")
def list_plans(db: Session = Depends(get_db)):
    return [{"id": p.id, "name": p.name, "description": p.description,
             "duration_days": p.duration_days}
            for p in db.query(RecoveryPlan).all()]


@router.post("/enroll/{plan_id}")
def enroll(plan_id: str, user: User = Depends(get_current_user),
           db: Session = Depends(get_db)):
    if not db.query(RecoveryPlan).filter(RecoveryPlan.id == plan_id).first():
        raise HTTPException(status_code=404, detail="Plan not found")
    pr = PatientRecovery(patient_id=user.id, plan_id=plan_id)
    db.add(pr)
    db.commit()
    db.refresh(pr)
    return {"patient_recovery_id": pr.id, "start_date": str(pr.start_date)}


@router.get("/my-tasks", response_model=list[RecoveryTaskOut])
def my_tasks(user: User = Depends(get_current_user), db: Session = Depends(get_db)):
    pr = (db.query(PatientRecovery)
          .filter(PatientRecovery.patient_id == user.id,
                  PatientRecovery.active == True)  # noqa: E712
          .first())
    if not pr:
        return []
    done = {p.recovery_task_id for p in db.query(RecoveryProgress)
            .filter(RecoveryProgress.patient_recovery_id == pr.id)}
    tasks = (db.query(RecoveryTask)
             .filter(RecoveryTask.plan_id == pr.plan_id)
             .order_by(RecoveryTask.day_number).all())
    return [RecoveryTaskOut(id=t.id, day_number=t.day_number, kind=t.kind,
                            instruction=t.instruction, completed=t.id in done)
            for t in tasks]


@router.post("/complete/{task_id}")
def complete_task(task_id: str, user: User = Depends(get_current_user),
                  db: Session = Depends(get_db)):
    pr = (db.query(PatientRecovery)
          .filter(PatientRecovery.patient_id == user.id,
                  PatientRecovery.active == True)  # noqa: E712
          .first())
    if not pr:
        raise HTTPException(status_code=404, detail="No active recovery plan")
    exists = (db.query(RecoveryProgress)
              .filter(RecoveryProgress.patient_recovery_id == pr.id,
                      RecoveryProgress.recovery_task_id == task_id).first())
    if not exists:
        db.add(RecoveryProgress(patient_recovery_id=pr.id, recovery_task_id=task_id))
        db.commit()
    return {"status": "completed", "task_id": task_id}
